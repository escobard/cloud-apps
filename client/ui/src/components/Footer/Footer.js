import React, { useContext } from "react";

import { useLocation } from "react-router-dom";

import { NotesContext } from "../../providers";

// TODO - fix constant absolute import to allow more readable imports
import { footer } from "../../constants/catalog";

import "./Footer.scss";

const Footer = () => {
  const { openModal, notes } = useContext(NotesContext);
  const { pathname } = useLocation();

  const hasNotes = notes && Array.isArray(notes) && notes.length >= 1;
  const isHome = pathname === "/";

  return (
    <footer>
      {hasNotes && isHome ? (
        <>
          <p>
            {footer.withNotes} <span> {notes.length} </span>
          </p>
        </>
      ) : (
        <p>{footer.noNotes}</p>
      )}
      {notes && Array.isArray(notes) && (
        <i
          aria-label="Add note"
          role="button"
          className="plus big icon"
          onClick={openModal}
          tabIndex={0}
          onKeyDown={(e) => {
            e.key === "Enter" && openModal();
          }}
        />
      )}
    </footer>
  );
};

export default Footer;
