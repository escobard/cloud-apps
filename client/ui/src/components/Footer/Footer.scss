@import "../../styles/variables";

footer{
  right: 0;
  left: 0;
  bottom: 0;
  border-top: solid 2px $gray-light;
  background: $white;
  text-align: center;
  position: fixed;
  margin: $space-xlrg $space-xlrg 0;
  padding: $space-xxlrg 0 $space-lrg;
  p{
    float: left;
    color: $gray-medium;
    font-weight: 500;
    margin-top: -3px !important;
    span{
      width: $space-lrg;
      display: block;
      float: right;
      margin-left: $space-small;
      background: $gray-medium-dark;
      color: $white;
      border-radius: 50%;
      font-size: $font-size-xs;
    }
  }
  i{
    float: right;
    height: 50px !important;
    background: $blue;
    width: 50px !important;
    border-radius: 50%;
    padding-top: 11.5px;
    color: $white;
    margin-top: -17px !important;
    &:hover{
      cursor: pointer;
    }
  }
}
