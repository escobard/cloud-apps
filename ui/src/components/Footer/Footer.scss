@import "../../styles/variables";

footer{
  right: 0;
  left: 0;
  bottom: 0;
  padding: $space-lrg 0;
  border-top: solid 2px $gray-light;
  background: $white;
  text-align: center;
  position: fixed;
  margin: 30px;
  p{
    float: left;
    span{
      width: $space-lrg;
      display: block;
      float: right;
      margin-left: $space-small;
      background: $gray-medium-dark;
      color: $white;
      border-radius: 50%;
    }
  }
}