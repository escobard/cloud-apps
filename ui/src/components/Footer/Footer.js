import React, { Fragment } from "react";
import PropTypes from "prop-types";

import { footer } from "constants/catalog";

import "./Footer.scss";

/** Footer Component
 * @param {number} count, number of notes
 * @param {function} open, parent function to open new note modal
 * @param {boolean} hasError, has errors
 * @return {Component} Footer
 * */

const Footer = ({ count, open, hasError }) => (
  <footer className={hasError ? "error" : undefined}>
    <p>
      {count >= 1 ? (
        <>
          {footer.withNotes} <span> {count} </span>
        </>
      ) : (
        <>{footer.noNotes}</>
      )}
    </p>
    <i role="button" className="plus big icon" onClick={open} />
  </footer>
);

Footer.propTypes = {
  count: PropTypes.number,
  open: PropTypes.func.isRequired,
  hasError: PropTypes.bool.isRequired
};

Footer.defaultProps = {
  count: 0
};

export default Footer;
