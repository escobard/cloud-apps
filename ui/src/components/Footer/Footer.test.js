import React from "react";
import { shallow } from "enzyme";
import Footer from "./index";

const props = {
  id: "test",
  count: 0,
  open: false
};

describe("Footer snapshot renders", () => {
  it("should render correctly mode", () => {
    const component = shallow(<Footer {...props} />);

    expect(component).toMatchSnapshot();
  });
});
