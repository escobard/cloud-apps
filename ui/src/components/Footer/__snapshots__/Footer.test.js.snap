// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer >> snapshot is up to date 1`] = `
<div>
  <footer
    class="error"
  >
    <p>
      NO NOTES
    </p>
    <i
      class="plus big icon"
      role="button"
    />
  </footer>
</div>
`;
